backend/dal/Services/Concrete/BuildingService.cs

```diff
@@ -161,7 +163,27 @@ public Building Add(Building building)
 
             building.AgencyId = agency.Id;
             building.Agency = agency;
+            building.Address.Province = this.Context.Provinces.Find(building.Address.ProvinceId);
+            building.Classification = this.Context.PropertyClassifications.Find(building.ClassificationId);
             building.IsVisibleToOtherAgencies = false;
+
+            building.Parcels.ForEach(bp =>
+            {
+                bp.Building = building;
+                //The Parcel may already exist, if it does update the existing parcel.
+                if (bp.ParcelId > 0)
+                {
+                    _service.Parcel.PendingUpdate(bp.Parcel);
```

Comment:
Will likely need to do something similar for the building update logic in a future PR.